### PR TITLE
Adding AMD shim api, issue #69

### PR DESCRIPTION
**Pull request by rorticus**
Adding support for the AMD shim API, as [outlined here](https://github.com/amdjs/amdjs-api/blob/master/CommonConfig.md#shim-).

Example,

```html
<script src="jquery.js"></script>
<script src="loader.js"></script>
<script>
require.config({
    shim: {
        'jquery': {
            exports: '$'
        }
    }
});

define(['jquery'], function (tool) {
    // tool holds $
});
</script>
```

### CODE REVIEW

**Comment by agubler** [comment]
@Tomdye can you have a review of this?

File: src/loader.ts [hunk 1]
```diff
@@ -292,6 +294,48 @@ declare const Packages: {} | undefined;
...
 			// Note that old paths will get destroyed if reconfigured
 			configuration.paths && (pathMapPrograms = computeMapProgram(configuration.paths));
+
+			// shim API
+			if (config.shim !== undefined) {
+				Object.keys(config.shim).forEach((moduleId) => {
+					let moduleDef: ModuleShim = (config.shim || {})[ moduleId ];
+
+					// using shorthand module syntax, convert to full syntax
+					if ('length' in moduleDef) {
```
Comment (Tomdye) [inline src/loader.ts]: assuming you're checking to see if this is an array, would `Array.isArray` not be better?

File: src/loader.ts [hunk 2]
```diff
@@ -292,6 +294,48 @@ declare const Packages: {} | undefined;
 
 			// Note that old paths will get destroyed if reconfigured
 			configuration.paths && (pathMapPrograms = computeMapProgram(configuration.paths));
+
+			// shim API
+			if (config.shim !== undefined) {
```
Comment (Tomdye) [inline src/loader.ts]: `if (config.shim)` should be enough as it will cover null cases too

File: src/loader.ts [hunk 3]
```diff
@@ -292,6 +294,48 @@ declare const Packages: {} | undefined;
...
+
+					// using shorthand module syntax, convert to full syntax
+					if ('length' in moduleDef) {
+						moduleDef = {
+							deps: <string[]> moduleDef
+						};
+					}
+
+					define(moduleId, moduleDef.deps || [], function (...dependencies) {
+						let root: any = undefined;
```
Comment (Tomdye) [inline src/loader.ts]: `let root` will already be undefined

File: src/loader.ts (hunk 1 above)
Comment (rorticus) [inline src/loader.ts]: Updated!

File: src/loader.ts (hunk 3 above)
Comment (rorticus) [inline src/loader.ts]: Updated!

**Pull request merged**